forge-game/src/main/java/forge/game/ability/effects/LifeLoseEffect.java

```diff
@@ -41,13 +46,19 @@ public void resolve(SpellAbility sa) {
 
         final int lifeAmount = AbilityUtils.calculateAmount(sa.getHostCard(), sa.getParam("LifeAmount"), sa);
 
+        final Map<Player, Integer> lossMap = Maps.newHashMap();
         for (final Player p : getTargetPlayers(sa)) {
             if (!p.isInGame()) {
                 continue;
             }
-            lifeLost += p.loseLife(lifeAmount, false, false);
+            final int lost = p.loseLife(lifeAmount, false, false);
+            lossMap.put(p, lost);
```

Comment:
We should really avoid running triggers if nothing happened which doesn't seem to be the case in all places

Reply:
> Yeah we probably need to think about multiple UnlessCost :/

Same for Lion's Eye Diamond